test_engine.py

```diff
@@ -0,0 +1,372 @@
+    assert isinstance(engine.val_dataloader, MonaiLoader)
+    assert isinstance(engine.test_dataloader, MonaiLoader)
+
+
+#########################################################################################################################
+def test_save_checkpoint(engine,ckpt_path='Checkpoint'):
+    """
+    Description: Tests the save_checkpoint method of the engine.
+
+    Parameters:
+
+      engine(object) : An instance of the Engine class.
+
+      ckpt_path (str): path of the input directory.
+
+    Returns: None
+    """
+
+    engine.save_checkpoint(ckpt_path) 
+
+    assert os.path.exists(ckpt_path)
+    
+#########################################################################################################################
+def test_load_checkpoint(engine,checkpoint_path = 'Checkpoint'):
+    """
+    Description:  Tests the load_checkpoint method of the engine.
+
+    Parameters:
+       
+      engine(object) : An instance of the Engine class.
+
+      checkpoint_path (str): path of the input directory.
+
+    Returns: None
+    """
+    get_transforms(engine) 
+
+    init_weights = engine.network.state_dict()
+
+    engine.load_checkpoint(checkpoint_path)
+
+    # Get loaded weights
+    loaded_weights = engine.network.state_dict()
+
+    # Check that weights match
+    for i in init_weights.keys():
+        assert torch.allclose(init_weights[i], loaded_weights[i])
+#########################################################################################################################
+
+def test_fit(engine,
+        epochs = 2, 
+        evaluate_epochs = 1,
+        batch_callback_epochs = 100,
+        save_weight =False,
+        save_path ='Checkpoint',
+
+        ):
+    """
+    Description: Tests the fit method of the engine.
+
+    Parameters:
+       
+        engine(object): An instance of the Engine class.
+
+        epochs (int): Number of training epochs.
+
+        evaluate_epochs (int): Number of epochs between model evaluation.
+
+        batch_callback_epochs (int): Number of epochs between batch-level callbacks.
+
+        save_weight (bool): Whether to save the model weights.
+
+        save_path (str): Path to save the model weights.
+
+    Returns: None
+    """
+    get_transforms(engine)
+    engine.load_data()
+
+    engine.fit(
+        epochs = epochs,
+        evaluate_epochs = evaluate_epochs,
+        batch_callback_epochs = batch_callback_epochs,
+        save_weight = save_weight,
+        save_path = save_path
+    )
+
+    assert os.path.exists(save_path)
+
+#########################################################################################################################
+def test_test(engine):
+    """
+    Description: Tests the test method of the engine.
+
+    Parameters:
+       
+      engine(object) : An instance of the Engine class.
+
+    Returns: None
+    """
+
+    get_transforms(engine)
+    engine.load_data()
+
+    loss, metric = engine.test()
+
+    assert loss != 0
```

Comment:
set a seed, then assert for the actual values

Reply:
done